src/controllers/update/beneficial.owner.type.controller.ts

```diff
@@ -42,16 +44,24 @@ export const get = async (req: Request, res: Response, next: NextFunction) => {
     const allBos = [
       ...(appData[BeneficialOwnerIndividualKey] ?? []),
       ...(appData[BeneficialOwnerOtherKey] ?? []),
-      ...(appData[BeneficialOwnerGovKey] ?? [])];
+      ...(appData[BeneficialOwnerGovKey] ?? [])
+    ];
 
-    const hasExistingBos = allBos.find(bo => bo.ch_reference) !== undefined;
     const hasNewlyAddedBos = allBos.find(bo => bo.ch_reference === undefined) !== undefined;
 
+    const allBosMos = [
+      ...allBos,
+      ...(appData[ManagingOfficerCorporateKey] ?? []),
+      ...(appData[ManagingOfficerKey] ?? [])
+    ];
+
+    const hasExistingBosMos = allBosMos.find(boMo => boMo.ch_reference) !== undefined;
```

Comment:
Suggest some() instead of find()

Reply:
also would be explicit about ch_reference !== undefined

Reply:
Functionality works with find() and it was the approach pushed for the last ticket